### PR TITLE
runtime-rs: remove attempt to access sandbox bundle from container bu…

### LINKED ISSUE #9847: runtime-rs: container start fails under cri-o
When cri-o tries to launch a runtime-rs pod it fails with
```
ERROR: containerd-shim-kata-v2 failed: shim start

Caused by:
   0: do start
   1: failed to get bundle path
   2: No such file or directory (os error 2): connect: invalid argument
```
This happens when the shim is launched with the "start" argument and `container_type` is `ContainerType::PodContainer` in [`ShimExecutor::do_start()`](https://github.com/kata-containers/kata-containers/blob/824287d64a9f030c7d796b34e330aba341a3feca/src/runtime-rs/crates/shim/src/shim_start.rs#L35).  [`ShimExecutor::get_shim_info_from_sandbox()`](https://github.com/kata-containers/kata-containers/blob/824287d64a9f030c7d796b34e330aba341a3feca/src/runtime-rs/crates/shim/src/shim_start.rs#L109) is then called which tries to derive the path to the sandbox bundle from the container bundle the shim is currently running in.

This fails since the bundle directory structure is different between containerd and cri-o (containerd stores the bundles in `<container_root>/<sandbox_id>` and `<container_root>/<container_id>` while cri-o in `<container_root>/<sandbox_id>/userdata` and `<container_root>/<container_id>/userdata`, respectively).

Having discussed this with @littlejawa we believe that the shim shouldn't try to access the sandbox bundle in the first place while starting a container.  For a smoke test, I've removed the offending code locally and have been running with it for a bit, having observed no ill effects so far.

Do you folks agree that this is the right solution?  Admittedly I'm not clear what the reason for the current shim behaviour was in the first place so I might be missing something.

### PR DESCRIPTION
**Pull request by pmores**
…ndle

Since no objections were raised in the linked issue (#9847) this commit removes the attempt to derive sandbox bundle path from container bundle path.  As described in more detail in the linked issue, this is container runtime specific and doesn't seem to serve any purpose.

Fixes #9847

### CODE REVIEW

File: src/runtime-rs/crates/shim/src/shim_start.rs [hunk 1]
```diff
@@ -105,22 +104,6 @@ impl ShimExecutor {
 
         Ok(child.id())
     }
-
-    fn get_shim_info_from_sandbox(&self, sandbox_id: &str) -> Result<(PathBuf, u32)> {
```
Comment (gkurz) [inline src/runtime-rs/crates/shim/src/shim_start.rs:109]: One user left at https://github.com/kata-containers/kata-containers/blob/e57e1122c9850816999efdc26ebdf2bc344e64fe/src/runtime-rs/crates/shim/src/shim_start.rs#L197.

`test_get_info_from_sandbox()` should be removed as well.

Comment (pmores) [reply src/runtime-rs/crates/shim/src/shim_start.rs:109]: Oh I missed that, good catch, thanks!

**Comment by littlejawa** [comment]
This is similar to what the go shim does: write the address, but not the PID.
Neither containerd nor crio have issue with the missing PID file when running the go shim, so I think this is OK.
Now I'd really like to hear from other runtime-rs developers to confirm that we're not missing something.

File: src/runtime-rs/crates/shim/src/shim_start.rs (hunk 1 above)
Comment (pmores) [reply src/runtime-rs/crates/shim/src/shim_start.rs:109]: This should be fixed now.

**Comment by pmores** [comment]
Force-pushed to placate clippy.  A bunch of required tests were failing but due to a k8s overconstrained allocation so that doesn't seem related to this change.

**Comment by gkurz** [comment]
> Force-pushed to placate clippy. A bunch of required tests were failing but due to a k8s overconstrained allocation so that doesn't seem related to this change.

Remaining failures seem to point to some instability in AKS. Hopefully this will auto-fix at some point.

**Review by gkurz — APPROVED** [review]
/lgtm

Thanks @pmores !

**Review by jepio — APPROVED** [review]
Not a runtime-rs developer but as far as i can tell this shouldn't cause any issues.

Do we have any CRI-O tests in the CI? If not - what's the plan to add some?

**Comment by gkurz** [comment]
> Not a runtime-rs developer but as far as i can tell this shouldn't cause any issues.
> 
> Do we have any CRI-O tests in the CI? If not - what's the plan to add some?

We have CRI-O tests in the CI but they are only run against the go runtime at this time. This fix might allow us to run them against runtime-rs as well.

**Comment by pmores** [comment]
> Do we have any CRI-O tests in the CI? If not - what's the plan to add some?

I'm quite sure there are CRI-O tests of the go shim but none of runtime-rs (quite obviously since doesn't even launch :-)).

The plan is to add some right now and it's being worked on actually (@wainersm @ldoktor ).  This is also why this fix is important - with cri-o being unable to launch runtime-rs no cri-o/runtime-rs CI testing is possible and no development and testing on OpenShift is possible either.

**Review by lifupan — APPROVED** [review]
LGTM

**Pull request merged**